src/modules/i18n/components/I18nProvider.js

```diff
@@ -75,11 +79,17 @@ class I18nProvider extends React.Component<PropsType, StateType> {
     }
   }
 
-  initLanguage () {
-    const targetLanguage = this.getI18nextLanguage()
+  async initLanguage () {
+    const { setUiDirection, setContentLanguage } = this.props
+    let language: ?string = await this.languageAsyncStorage.loadLanguage()
+    if (!language) {
+      language = this.getI18nextLanguage()
+      await this.languageAsyncStorage.setLanguage(language)
+    }
 
-    this.setState({language: targetLanguage})
-    this.props.setUiDirection(RTL_LANGUAGES.includes(targetLanguage) ? 'rtl' : 'ltr')
+    this.setState({ language })
+    setUiDirection(RTL_LANGUAGES.includes(language) ? 'rtl' : 'ltr')
```

Comment:
These two lines affect the UI language. The content in the languageAsyncStorage (the content language) should not affect the ui language.